github/code-scanning.go

```diff
@@ -18,6 +18,51 @@ import (
 // GitHub API docs: https://docs.github.com/en/free-pro-team@latest/rest/reference/code-scanning/
 type CodeScanningService service
 
+// Rule represents the complete details of GitHub Code Scanning alert type.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository
+type Rule struct {
+	ID              *string  `json:"id,omitempty"`
+	Severity        *string  `json:"severity,omitempty"`
+	Description     *string  `json:"description,omitempty"`
+	Name            *string  `json:"name,omitempty"`
+	FullDescription *string  `json:"full_description,omitempty"`
+	Tags            []string `json:"tags,omitempty"`
+	Help            *string  `json:"help,omitempty"`
+}
+
+// Location represents the exact location of the GitHub Code Scanning Alert in the scanned project.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository
+type Location struct {
+	Path        *string `json:"path,omitempty"`
+	StartLine   *int    `json:"start_line,omitempty"`
+	EndLine     *int    `json:"end_line,omitempty"`
+	StartColumn *int    `json:"start_column,omitempty"`
+	EndColumn   *int    `json:"end_column,omitempty"`
+}
+
+// Message is a part of MostRecentInstance struct which provides the appropriate message when any action is performed on the analysis object
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository
```

Comment:
Since all these struct URLs are identical, let's just please delete them to avoid the duplication.  Thanks.